src/transformer/utils/extensions/base.php

```diff
@@ -17,7 +17,7 @@
 namespace src\transformer\utils\extensions;
 defined('MOODLE_INTERNAL') || die();
 
-function base(array $config, \stdClass $event, $course) {
+function base(array $config, \stdClass $event, $course=null) {
```

Comment:
I'm curious about the reason for this change.

Reply:
I've used `base` in `src/transformer/events/totara_program/program_assigned.php` in a context where a course doesn't exist. The course parameter is only required in this function for the jisc data.

In `src/transformer/utils/extensions/course_area.php` the function `course_area` (called from jisc util) already has a condition for if the course is null. I expect my change was actually how it was intended to be initially.

Reply:
Makes sense thanks for explaining